tests/util/test_recursive_replace.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import annotations
+
+import copy
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Union
+
+import pytest
+
+from chia.util.recursive_replace import recursive_replace
+
+
+class TestC:
+    a: int
+    b: str
+
+    def __init__(self, a: int, b: str):
+        self.a = a
+        self.b = b
+
+    def replace(self, **kwargs: Dict[str, Union[int, str, Optional[TestA]]]) -> TestC:
+        ret = TestC(copy.deepcopy(self.a), copy.deepcopy(self.b))
+        for key, value in kwargs.items():
+            if key == "a":
+                ret.a = value  # type: ignore[assignment]
+            elif key == "b":  # pragma: no cover
+                ret.b = value  # type: ignore[assignment]
+            else:  # pragma: no cover
+                raise TypeError(f"unknown field {key}")
+        return ret
+
+
+@dataclass
+class TestA:
+    a: int
+    b: str
+    c: List[int]
+    d: Optional[TestC]
+
+
+class TestB:
+    a: int
+    b: str
+    c: Optional[TestA]
+
+    def __init__(self, a: int, b: str, c: Optional[TestA]):
+        self.a = a
+        self.b = b
+        self.c = c
+
+    def replace(self, **kwargs: Dict[str, Union[int, str, Optional[TestA]]]) -> TestB:
```

Comment:
I get that this is kinda meant as a stand-in for rust implementations and as such the actual implementation doesn't matter.  Would probably be good to add a note warning about this fact and how this is not a reference implementation, just a test implementation.

If this isn't the case, I'll give a real review with various suggestions about patterns to use here.

Reply:
it *is* just a way to test the non-dataclass path in `recursive_replace()`. As such, I think it makes sense to keep it minimal and not necessarily robust for real-world use.